src/indra_cogex/analysis/source_targets_explanation.py

```diff
@@ -313,249 +311,299 @@ def get_go_terms_for_source(source_hgnc_id):
     return source_go_terms, go_nodes
 
 
-def shared_upstream_bioentities_from_targets(stmts_by_protein_df, filename):
-    """Get a dataframe of upstream molecules intersecting with bioentities that target proteins with direct INDRA
-       relationships
+@autoclient()
+def shared_upstream_bioentities_from_targets(
+    stmts_by_protein_df: pd.DataFrame,
+    target_genes: List[str],
+    *,
+    client
+) -> Tuple[List[str], pd.DataFrame]:
+    """Get upstream molecules intersecting with bioentities.
 
     Parameters
     ----------
-    filename : str
-        Path to the CSV file containing upstream bioentities for gene sets.
-    stmts_by_protein_df : dataframe
-        Contains all bioentities target protien has a direct INDRA relationship
+    stmts_by_protein_df : pd.DataFrame
+        DataFrame containing INDRA statements
+    target_genes : List[str]
+        List of target gene symbols
+    client :
```

Comment:
Add type for the client parameter or add this to the docstring:
```suggestion
    client : Neo4jClient
```

Reply:
FIxed in 3b58c073